src/client/Microsoft.Identity.Client/Http/DefaultRetryPolicy.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.Identity.Client.Core;
+
+namespace Microsoft.Identity.Client.Http
+{
+    class DefaultRetryPolicy : IRetryPolicy
```

Comment:
nit: `Default` is a pretty overloaded term. Let's use `StsEndpointsRetryPolicy` maybe?

Reply:
I like Default, because this is used by both ESTS **AND** Managed Identity sources without any specific policy

Reply:
I don't think this will be used by managed identity sources at all. Its either the ones specific to managed identity or NoRetryPolicy.

Reply:
@neha-bhargava, that's incorrect as I understand it. All Managed Identities default to the DefaultRetryPolicy unless otherwise specified. This is how it works in msal-node.

Reply:
@neha-bhargava - I thought all MSI sources use a linear 3 time retry, with 1 second pause? While the STS uses 1 retrty.

Reply:
@Robbie-Microsoft - I want to challenge this `default` concept again. Relying on a "default" behavior will lead MSAL developers to make mistakes whenever we call new endpoints. We'll likely overuse "default". It's also complicated, as the "default" has a different shape between MSI and STS. But I realize I have looked too much at this and my opinion is tainted.

@neha-bhargava @gladjohn - can you please provide a perspective on this one?

Reply:
@bgavrilMS you are right. All MSI does 3 retry with a linear delay of 1 sec and ests 1 retry with 1 sec delay.

Reply:
@bgavrilMS @neha-bhargava agree the only difference I've know is the 1 second pause between them. But I think @Robbie-Microsoft did this way because there isn't a one second retry in the MSI flows. I filed a task to add it.

Reply:
https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/5260